### ISSUE TITLE
Failed to connect to sparkpost smtp

### ISSUE THREAD

**Issue by robbrandt** [PHPMailer/PHPMailer#1522]
# Problem description

connect() failed.  This is an existing system that suddenly started failing. I've upgraded to the latest 5.2.26 version with no improvement.

# Code to reproduce

$mail = new PHPMailer();
	$mail->CharSet = 'UTF-8';
	$mail->Encoding = "base64";
	$mail->IsSMTP(true); // telling the class to use SMTP
	$mail->SMTPDebug = 2;
	$mail->Timeout  = 30;
	$mail->Host       = "smtp.sparkpostmail.com"; // SMTP server
	$mail->SMTPAuth   = true;                  // enable SMTP authentication
	$mail->Host       = "smtp.sparkpostmail.com"; // SMTP server
	$mail->Port       = 587;                    // set the SMTP port for the GMAIL server
	$mail->SMTPSecure = 'tls';
	$mail->Username   = "SMTP_Injection"; // SMTP account username
	$mail->Password   = "xxxxxxxxxxx......";        // SMTP account password
	$mail->SetFrom($maincontact, $maincontactname);

	$mail->Subject = "$sitename Proposal Submission Copy #".$_SESSION['newAbstractID']."";
	$mail->Body = html_entity_decode($copyEmail, ENT_COMPAT, 'UTF-8');
	$mail->WordWrap = 50;
        $mail->AddAddress($_SESSION["email"],"".$_SESSION["fullname"]."");;
	$mail->AddCC($maincontact,$maincontactname);

# Debug output
2018-08-15 18:55:46 SERVER -> CLIENT: 220 2.0.0 smtp.sparkpostmail.com ESMTP ecelerity 4.2.39.65443 r(Core:4.2.39.51) Wed, 15 Aug 2018 18:55:46 +0000
2018-08-15 18:55:46 CLIENT -> SERVER: EHLO localhost
2018-08-15 18:55:46 SERVER -> CLIENT: 250-a.mta1vsmtp.cc.aws-usw2a.prd.sparkpost says EHLO to 68.6.112.199:57250250-ENHANCEDSTATUSCODES250-PIPELINING250-AUTH=LOGIN PLAIN250-AUTH LOGIN PLAIN250-STARTTLS250 8BITMIME
2018-08-15 18:55:46 CLIENT -> SERVER: STARTTLS
2018-08-15 18:55:46 SERVER -> CLIENT: 220 2.0.0 continue
SMTP Error: Could not connect to SMTP host.
2018-08-15 18:55:46 SMTP NOTICE: EOF caught while checking if connected
SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting

**Comment by Synchro** [comment]
Your TLS connection is failing immediately after STARTTLS. This problem is described exactly in the troubleshooting guide that the error message links to. Please read that and do what it says. It’s most likely that your CA certificates are outdated and need updating.

**Comment by robbrandt** [comment]
Thank you for the explanation.  It wasn't at all clear to me from the debug output and the Troubleshooting page that CA certificates were the problem.

But it seems more complicated than this. Using the 
echo QUIT | openssl s_client -crlf -starttls smtp -CAfile /etc/ssl/cacert.pem -connect smtp.gmail.com:587
command shows that there is indeed a problem, yet my OS is completely up to date and even using the specific update-ca-certificates command says everything is updated.  This is Ubuntu 14.04.  Yet others report similar problems:
https://askubuntu.com/questions/646594/how-to-fix-ca-cert-issues-with-curl-in-ubuntu-14-04

I am still working through this...

**Comment by robbrandt** [comment]
So it seems I have everything up to date now, getting a good result from 
    echo QUIT | openssl s_client -crlf -starttls smtp -CAfile /etc/ssl/cacert.pem -connect smtp.gmail.com:587

What seemed to work was downloading a fresh .pem from http://curl.haxx.se/ca/cacert.pem.  I also did update-ca-certificates --fresh but that didn't seem to improve things.

However mail still cannot be sent.  I tried adding
    openssl.cafile = /etc/ssl/cacert.pem
    curl.cainfo = /etc/ssl/cacert.pem
to my php.ini but I am still running php 5.9.x, and those directives are for 5.6.  (this is a server running some older apps not updated to mysqli, and that's the last version that "works").

Debug output is still the same:
2018-08-15 22:29:42 SERVER -> CLIENT: 220 2.0.0 smtp.sparkpostmail.com ESMTP ecelerity 4.2.39.65443 r(Core:4.2.39.51) Wed, 15 Aug 2018 22:29:42 +0000
2018-08-15 22:29:42 CLIENT -> SERVER: EHLO localhost
2018-08-15 22:29:42 SERVER -> CLIENT: 250-b.mta1vsmtp.cc.aws-usw2b.prd.sparkpost says EHLO to 54.225.91.96:41019250-PIPELINING250-STARTTLS250-AUTH=LOGIN PLAIN250-AUTH LOGIN PLAIN250-ENHANCEDSTATUSCODES250 8BITMIME
2018-08-15 22:29:42 CLIENT -> SERVER: STARTTLS
2018-08-15 22:29:42 SERVER -> CLIENT: 220 2.0.0 continue
SMTP Error: Could not connect to SMTP host.
2018-08-15 22:29:42 SMTP NOTICE: EOF caught while checking if connected
SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting
